org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java

```diff
@@ -69,6 +75,39 @@ private static Filter createFilter() {
 		return context.registerService(ApplicationLauncher.class, new BndApplicationLauncher(context), null);
 	}
 
+	private void setEquinoxConfigAppArgs(BundleContext context) {
+		String[] args = null;
+		try {
+			ServiceReference<?>[] serviceReferences = context.getServiceReferences("aQute.launcher.Launcher",
+				"(launcher.arguments=*)");
+			if (serviceReferences != null && serviceReferences.length > 0) {
+				Object property = serviceReferences[0].getProperty("launcher.arguments");
+
+				if (property instanceof String[]) {
+					args = (String[]) property;
+
+					ServiceReference<EnvironmentInfo> srvcRef = context.getServiceReference(EnvironmentInfo.class);
+					if (srvcRef != null) {
+						EnvironmentInfo service = context.getService(srvcRef);
+						if (service instanceof EquinoxConfiguration) {
+							EquinoxConfiguration equinoxConfig = (EquinoxConfiguration) service;
+							equinoxConfig.setAllArgs(args);
+							equinoxConfig.setAppArgs(args);
```

Comment:
Why are the command line args (`setAllArgs`) and the non-framework args (`setAppArgs`) set to the same value?

Reply:
AppArgs is all I need so I'll remove the other.